lib/bundler/plugin.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+module Bundler
+  module Plugin
+    autoload :Api,        "bundler/plugin/api"
+    autoload :Dsl,        "bundler/plugin/dsl"
+    autoload :Index,      "bundler/plugin/index"
+    autoload :Installer,  "bundler/plugin/installer"
+    autoload :SourceList, "bundler/plugin/source_list"
+
+    PLUGIN_FILE_NAME = "plugin.rb".freeze
+
+    @commands = {}
+
+    class << self
```

Comment:
since this is never instantiated, maybe it should be a module with `module_function` instead?